src/app/reporting/ReportSchedulerImpl.h

```diff
@@ -33,29 +53,98 @@ class ReportSchedulerImpl : public ReportScheduler
     ~ReportSchedulerImpl() override { UnregisterAllHandlers(); }
 
     // ICDStateObserver
+
+    /**
+     * @brief When the ICD changes to Idle, no action is taken in this implementation.
+     */
+    void OnTransitionToIdle() override{};
+
+    /**
+     * @brief When the ICD changes to Active, this implementation will trigger a report emission on each ReadHandler that is not
+     * blocked on its min interval.
+     *
+     * @note Most action triggering a change to the Active mode already trigger a report emission, so this method is optionnal as it
+     * might be redundant.
+     */
     void OnEnterActiveMode() override;
-    void OnTransitionToIdle() override;
-    // No action is needed by the ReportScheduler on ICD operation Mode changes
+
+    /**
+     * @brief When the ICD changes operation mode, no action is taken in this implementation.
+     */
     void OnICDModeChange() override{};
 
     // ReadHandlerObserver
+
+    /**
+     * @brief When a ReadHandler is added, adds a node and register it in the scheduler node pool. Scheduling the report here is
+     * un-necessary since the ReadHandler will call MoveToState(HandlerState::CanStartReporting);, which will call
+     * OnBecameReportable() and schedule the report.
+     *
+     * @note This method sets a now Timestamp that is used to calculate the next report timeout.
+     */
     void OnSubscriptionEstablished(ReadHandler * aReadHandler) final;
+
+    /**
+     * @brief When a ReadHandler becomes reportable, recalculate and reschedule the report.
+     *
+     * @note This method sets a now Timestamp that is used to calculate the next report timeout.
+     */
     void OnBecameReportable(ReadHandler * aReadHandler) final;
+
+    /**
+     * @brief When a ReadHandler report is sent, recalculate and reschedule the report.
+     *
+     * @note This method is called after the report is sent, so the ReadHandler is no longer reportable, and thus CanBeSynced and
+     * EngineRunScheduled of the node associated to the ReadHandler are set to false in this method.
+     *
+     * @note This method sets a now Timestamp that is used to calculate the next report timeout.
+     */
     void OnSubscriptionReportSent(ReadHandler * aReadHandler) final;
+
+    /**
+     * @brief When a ReadHandler is destroyed, remove the node from the scheduler node pool and cancel the timer associated to it.
+     */
     void OnReadHandlerDestroyed(ReadHandler * aReadHandler) override;
 
     virtual bool IsReportScheduled(ReadHandler * aReadHandler);
 
     void ReportTimerCallback() override;
 
 protected:
+    /**
+     * @brief Schedule a report for the ReadHandler associated to the node.
+     *
+     *  If a report is already scheduled for the ReadHandler, cancel it and schedule a new one.
+     *  If the timeout is 0, directly calls the TimerFired() method of the node instead of scheduling a report.
```

Comment:
What is the audience for this comment?  This seems to be a restatement of the method's implementation, but why does anyone care (who would not just read the implementation)?

I thought the goal here, based on what I had asked for, was documentation of how the system works as a whole, not per-method restatements of what the code in those methods does...

Reply:
The audience of the second part would be someone trying to establish if a report is scheduled by using the IsReportScheduled() method, which looks at the timer to assess if it is the case.

This will be moved to the IsReportScheduler method above.

The first part is implementation and will be move there.